city_scrapers/spiders/il_sex_offender_management.py

```diff
@@ -0,0 +1,249 @@
+        """Parse end datetime as a naive datetime object."""
+        date_str = self._get_meeting_date(clean_text)
+        time_str = self._get_start_end_time(clean_text)
+        # if meeting is cancelled or no time is provided, default is 12:00am
+        if time_str is None:
+            return self._get_datetime_obj(date_str, "12:00am")
+        return self._get_datetime_obj(date_str, time_str[7:])
+
+    def _get_meeting_date(self, clean_text):
+        """Parse meeting agenda for 'Month Day, Year' and returns group"""
+        pattern = re.compile(
+            r"(Jan(uary)?|Feb(ruary)?|Mar(ch)?|Apr(il)?|May|Jun(e)?|"
+            r"Jul(y)?|Aug(ust)?|Sep(tember)?|Oct(ober)?|Nov(ember)?|"
+            r"Dec(ember)?)\s+\d{1,2},\s+\d{4}"
+        )
+        return pattern.search(clean_text).group()
+
+    def _get_start_end_time(self, clean_text):
+        """Parse start and end time of meeting from agenda and returns group"""
+        pattern = re.compile(
+            r"[0-2]?[0-9]:[0-9][0-9][a-zA-Z][a-zA-Z]-"
+            r"[0-2]?[0-9]:[0-9][0-9][a-zA-Z][a-zA-Z]"
+        )
+        if pattern.search(clean_text) is not None:
+            return pattern.search(clean_text).group()
+        else:
+            # in a few specific agendas there are times formatted as '9-10am'
+            # instead of 9:00am-10:00am or 09:00am-10:00am
+            pattern = re.compile(r"[0-2]?[0-9]-[0-9]?[0-9][a-zA-Z][a-zA-Z]")
+            if pattern.search(clean_text) is not None:
+                # re-make string to 09:00am-10:00am format from '9-10am'
+                new_string = pattern.search(clean_text).group()
+                suffix = new_string[-2:]
+                start = new_string.split("-")[0] + ":00" + suffix
+                end = new_string.split("-")[1][:-2] + ":00" + suffix
+                new_string = start + "-" + end
+                return new_string
+            else:
+                return None
+
+    def _get_datetime_obj(self, date_str, time_str):
+        """Return Naive Datetime object from date and time strings passed in"""
+        pattern = re.compile(r"[0-2]?[0-9]:[0-9][0-9][a-zA-Z][a-zA-Z]")
+        time_str = pattern.search(time_str).group()
+        # standardize time by converting to military time
+        time_str = datetime.strptime(time_str, "%I:%M%p").strftime("%H:%M")
+        return datetime.strptime("{} {}".format(date_str, time_str), "%B %d, %Y %H:%M")
+
+    def _parse_location(self, clean_text):
+        """Parse or generate location."""
+        # there were different locations depending on if the meeting was
+        # a video conference, a phone conference, or a physical location.
+        # I hardcoded in the most frequent addresses as there were quite a few
+        # different locations, but I have made sure to note that one should conirm
+        # the address is indeed correct before using.
+
+        name = ""
+        address = ""
+
+        if "video conference" in clean_text.lower():
```

Comment:
If the location changes this much we can leave `address` blank and replace `name` with "See meeting details"